src/main/java/com/bigchaindb/builders/BigchainDbTransactionBuilder.java

```diff
@@ -326,11 +340,19 @@ public IBuild build(EdDSAPublicKey publicKey) {
                 this.transaction.addInput(input);
             }
 
-            if (this.transaction.getOperation() == null) {
+            if (this.operation == null) {
                 this.transaction.setOperation("CREATE");
+            } else {
+                this.transaction.setOperation(this.operation.name());
             }
 
-            this.transaction.setAsset(new Asset(this.assets, this.assetsDataClass));
+            if (String.class.isAssignableFrom(this.assets.getClass())) {
+                // interpret as an asset ID
+                this.transaction.setAsset(new Asset((String) this.assets));
+            } else {
+                // otherwise it's an asset
+                this.transaction.setAsset(new Asset(this.assets, this.assetsDataClass));
+            }
```

Comment:
this allows passing in an `Asset` `id` with this kind of call:

`addAsset("id_of_asset", String.class)`

If an asset is a `String`, it is considered an ID. i am open to feedback here if the approach is problematic

Reply:
nice. I had similar implementation in fix from my side 👍